app/app.go

```diff
@@ -165,266 +183,299 @@ type App struct {
+	app.GovKeeper.SetLegacyRouter(govv1beta1.NewRouter())
 
+	// Create packet forward keeper
 	app.PacketForwardKeeper = packetforwardkeeper.NewKeeper(
-		appCodec,
+		encodingConfig.Codec,
 		keys[packetforwardtypes.StoreKey],
-		app.GetSubspace(packetforwardtypes.ModuleName),
 		app.TransferKeeper, // will be zero-value here, reference is set later on with SetTransferKeeper.
 		app.IBCKeeper.ChannelKeeper,
-		app.DistrKeeper,
 		app.BankKeeper,
-		tokenFilterKeeper,
+		app.IBCKeeper.ChannelKeeper,
+		govModuleAddr,
 	)
 
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
-		appCodec, keys[ibctransfertypes.StoreKey], app.GetSubspace(ibctransfertypes.ModuleName),
-		app.PacketForwardKeeper, app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
-		app.AccountKeeper, app.BankKeeper, app.ScopedTransferKeeper,
+		encodingConfig.Codec, keys[ibctransfertypes.StoreKey], app.GetSubspace(ibctransfertypes.ModuleName),
+		app.PacketForwardKeeper, app.IBCKeeper.ChannelKeeper, app.IBCKeeper.PortKeeper,
+		app.AccountKeeper, app.BankKeeper, app.ScopedTransferKeeper, govModuleAddr,
 	)
 	// Transfer stack contains (from top to bottom):
 	// - Token Filter
 	// - Packet Forwarding Middleware
 	// - Transfer
 	var transferStack ibcporttypes.IBCModule
 	transferStack = transfer.NewIBCModule(app.TransferKeeper)
-	packetForwardMiddleware := packetforward.NewIBCMiddleware(
-		transferStack,
-		app.PacketForwardKeeper,
+	transferStack = packetforward.NewIBCMiddleware(transferStack, app.PacketForwardKeeper,
 		0, // retries on timeout
 		packetforwardkeeper.DefaultForwardTransferPacketTimeoutTimestamp, // forward timeout
-		packetforwardkeeper.DefaultRefundTransferPacketTimeoutTimestamp,  // refund timeout
 	)
-	// PacketForwardMiddleware is used only for version >= 2.
-	transferStack = module.NewVersionedIBCModule(packetForwardMiddleware, transferStack, v2, v3)
+
 	// Token filter wraps packet forward middleware and is thus the first module in the transfer stack.
-	tokenFilterMiddleware := tokenfilter.NewIBCMiddleware(transferStack)
-	transferStack = module.NewVersionedIBCModule(tokenFilterMiddleware, transferStack, v1, v3)
-
-	app.EvidenceKeeper = *evidencekeeper.NewKeeper(
-		appCodec,
-		keys[evidencetypes.StoreKey],
-		&app.StakingKeeper,
-		app.SlashingKeeper,
-	)
+	transferStack = tokenfilter.NewIBCMiddleware(transferStack)
 
-	app.GovKeeper = govkeeper.NewKeeper(
-		appCodec,
-		keys[govtypes.StoreKey],
-		app.GetSubspace(govtypes.ModuleName),
-		app.AccountKeeper,
-		app.BankKeeper,
-		&stakingKeeper,
-		govRouter,
-		baseApp.MsgServiceRouter(),
-		govtypes.DefaultConfig(),
+	// create evidence keeper with router
+	evidenceKeeper := evidencekeeper.NewKeeper(
+		encodingConfig.Codec, runtime.NewKVStoreService(keys[evidencetypes.StoreKey]), app.StakingKeeper, app.SlashingKeeper, app.AccountKeeper.AddressCodec(), runtime.ProvideCometInfoService(),
 	)
+	// If evidence needs to be handled for the app, set routes in router here and seal
```

Comment:
[nit] this comment seems helpful when originally wiring the `app.go` but seems less relevant for reviewers reading this code _after_ things have been wired together correctly.

TBH it doesn't seem relevant anymore because AFAICT we're not setting routes on the evidence keeper in the lines following so seems like a distraction

```suggestion
```